### PR TITLE
Binary loader for Assimp files

### PR DESCRIPTION
**Pull request by Virtulous**
Hey Everyone,

First off, I know how much you guys hate maintaining loaders, so I won't be offended if you don't want to merge this. 

That being said, I'd like to get it out there for people to see. We use this at [Virtulous](https://virtulo.us/) and have had a lot of success with it.  You can see a big list of successful imports at our [Model Repository](https://virtulo.us/1/3dr/ui/models/all/0). Create an account to upload a model to test the conversion and parsing. 

I see there is already a JSON format for Assimp, but there are some major downsides with using JSON to represent models. We really needed a stable, mature and efficient model format for delivering skinned and animated meshes to the browser. We looked at GLTF, but the support seems spotty. 

The benefit of using the native Assimp Binary is that there is already a mature and stable converter for a lot of different formats. 

This loader supports just about every mesh attribute that Assimp does. We currently don't import lights or cameras, since these are not used in our engine. 

- Skinning
- Animation
- Multiple UVs
- Bump and Normal maps
- Vertex Colors

This thing could use some improvements. In particular, it would be nice to translate our internal Virtulous.Animation into the native Three.js formats. If anyone would like to do that, you can submit a PR against [AssimpBinaryLoader](https://github.com/Virtulous/AssimpBinaryLoader)

I've done my best to get this into the Mr.Doob approved code style. We use a different style in house, so this conversion is probably not complete. Sorry about that. I tried that online formatter tool, but had no luck getting it to work. If you had a Sublime jsFormat config file that does it correctly, I'd love to have that.

### CODE REVIEW

**Comment by mrdoob** [comment]
I think this is great! How about just calling it `AssimpLoader` for clarity?

**Comment by Virtulous** [comment]
Sure, like so?

**Comment by mrdoob** [comment]
Yep. I guess you can remove ABINloader.js from the PR though?

File: examples/webgl_loader_assimp.html [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    	var controls = new THREE.OrbitControls( camera, renderer.domElement );
+		controls.target.set(-100,0,0);
+		
+		controls.update();
+
+    	stats = new Stats();
+    	container.appendChild( stats.dom );
+    	var loader = new THREE.AssimpLoader();
+
+    	loader.load( "./models/assbin/squid.assbin", function( err, object ) {
```
Comment (mrdoob) [inline examples/webgl_loader_assimp.html]: I would also rename `models/assbin` to `models/assimp`.

**Comment by Virtulous** [comment]
Ahh geez, sorry about that. I always forget you must `git rm` the old one and `git add` the new one in the same commit for it to count as a rename.

**Comment by mrdoob** [comment]
Sweet! Thanks!

**Comment by mrdoob** [comment]
I guess the source file is `Octaminator_lowpoly_final.fbx` inside http://opengameart.org/sites/default/files/Octaminator_0.zip?

**Comment by Virtulous** [comment]
Yes, that was the original input into the assimp commandline tool. I'm using Assimp 3.2 on Windows, but I don't think 3.3 or Linux should make any difference.

**Comment by mrdoob** [comment]
Okey, I've reformated the whole file... https://github.com/mrdoob/three.js/commit/01e32f5e44ab712190a8fdb8c52f5ba6913dc2d5?w=1 There are still some issues (mostly undefined variables) but, do you mind trying it with your files to check that I didn't break anything?

https://github.com/mrdoob/three.js/blob/dev/examples/js/loaders/AssimpLoader.js

**Comment by Virtulous** [comment]
Sure, give me a day to build a test harness, and I'll run all 300 some sample files.

**Pull request merged**